src/main/java/dan200/computercraft/core/apis/TermAPI.java

```diff
@@ -101,6 +111,59 @@ public static int parseColour( Object[] args, boolean _enableColours ) throws Lu
         };
     }
 
+    public static void setColour( Terminal terminal, int colour, float r, float g, float b )
+    {
+        if( terminal.getPalette() != null )
+        {
+            terminal.getPalette().setColour( colour, r, g, b );
+            terminal.setChanged();
+        }
+    }
+
+    public static void setColour( Terminal terminal, HashMap<Object, Object> colours) throws LuaException
```

Comment:
Setting 16 colours only required 256 writes because of the redundant setting of all 16 colours on each setColour call in window. Now that it only sets one, it's back to being a linear relationship so should be fine.